### DISCUSSION TITLE
gl_FragData

### DISCUSSION THREAD

**Discussion by jeanlemotan** [google/filament#4421] [Q&A] [answered]
If I try to use gl_FragData in a shader I get this compilation error:
0:13: 'gl_FragData' : undeclared identifier
0:13: 'gl_FragData' :  left of '[' is not of type array, matrix, or vector
0:13: '' : compilation terminated
3 compilation errors.  No code generated.

I'm trying to bind a render target with 2 color attachments and write in the second one some custom data (a render ID per object)

I couldn't find any sample, issue or discussion about this so I'm inclined to thing that it's not supported, but on the other hand the RenderTarget does support multiple color attachments.
Am I doing something obviously wrong?

This is the line:
`gl_FragData[2] = vec4(0.2, 0.0, 0.1, 1.0);`

//Edit - tried to get the formatting of the error fixed

**Comment by romainguy** [comment, marked as answer]
We currently do not (entirely) support MRT on the user side. We use them internally but we haven't exposed the APIs just yet. Note that when we do, it won't use `gl_FragData`, instead the materials declare multiple named outputs as expected in a modern GL setting.

**Reply by jeanlemotan** [reply]
Ok, that explains it. I just saw the proper declaration in the shader:
`layout(location = 0) out vec4 fragColor;`

I'll wait for official support and in the mean time I'll try to do multiple passes to generate the data I need.

Thanks!

**Reply by pixelflinger** [reply]
This is supported (officially). You can see an example in the ibl-prefilter library, which is just a client of filament.
Currently you're limited to using a post process material though.

**Reply by romainguy** [reply]
@pixelflinger Did we end up exposing post processing materials? I somehow thought they were still for internal use only. If so, great news :)

**Reply by jeanlemotan** [reply]
Got it, so in post I can use MRT but not in regular shaders. 
I've seen that internally filament uses MRT for the color correction pass when framebuffer fetch is supported to keep things on tile and avoid storing the HDR render target. Neat.

Thanks!
